### PR TITLE
Xray export

### PR DESCRIPTION
**Pull request by junkerm**
[Trello Card]()

https://trello.com/c/AVpi7HAG/514-testprozeduren-nach-xray-exportieren
https://trello.com/c/loMBY6fb/515-berechtigungspr%C3%BCfung-f%C3%BCr-xray

### CODE REVIEW

File: bundles/specmate-dummy-data/src/com/specmate/dummydata/DummyProject.java [hunk 1]
```diff
@@ -65,9 +66,21 @@ public boolean isAuthorizedToExport(String username, String password) {
...
+
+			@Override
+			public boolean canExportTestProceure() {
+				// TODO Auto-generated method stub
+				return false;
+			}
+
+			@Override
+			public boolean canExportTextSpecification() {
+				// TODO Auto-generated method stub
```
Comment (sebeder) [inline bundles/specmate-dummy-data/src/com/specmate/dummydata/DummyProject.java]: Remove auto-generated `TODO` comments.

File: bundles/specmate-dummy-data/src/com/specmate/dummydata/DummyProject.java [hunk 2]
```diff
@@ -65,9 +66,21 @@ public boolean isAuthorizedToExport(String username, String password) {
...
 
 			@Override
-			public void export(TestProcedure testProcedure) throws SpecmateException {
+			public void export(EObject target) throws SpecmateException {
 				// Do nothing
 			}
+
+			@Override
+			public boolean canExportTestProceure() {
+				// TODO Auto-generated method stub
```
Comment (sebeder) [inline bundles/specmate-dummy-data/src/com/specmate/dummydata/DummyProject.java]: Remove auto-generated `TODO` comments.

File: bundles/specmate-jira-connector/src/com/specmate/connectors/jira/internal/services/JiraExportService.java [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+		} finally {
+			if (jiraClient != null) {
+				try {
+					jiraClient.close();
+				} catch (IOException e) {
+					logService.log(LogService.LOG_ERROR, "Could not close jira client", e);
+				}
+			}
+		}
+// This is not working:
```
Comment (sebeder) [inline bundles/specmate-jira-connector/src/com/specmate/connectors/jira/internal/services/JiraExportService.java]: Remove commented code.

File: bundles/specmate-jira-connector/src/com/specmate/connectors/jira/internal/services/JiraUtil.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import com.atlassian.jira.rest.client.api.RestClientException;
+import com.atlassian.jira.rest.client.internal.async.AsynchronousJiraRestClientFactory;
+import com.specmate.common.exception.SpecmateAuthorizationException;
+import com.specmate.common.exception.SpecmateException;
+
+public class JiraUtil {
+
+	public static JiraRestClient createJiraRESTClient(String url, String username, String password)
+			throws URISyntaxException {
+		// curl -s -H "Authorization: Basic ***REMOVED***"
```
Comment (sebeder) [inline bundles/specmate-jira-connector/src/com/specmate/connectors/jira/internal/services/JiraUtil.java]: That's from me ;-) Please remove it.

**Review by sebeder — CHANGES_REQUESTED** [review]
Hi Max, I'll still need to test it at our customer. However, there are some minor issues with the code.

File: bundles/specmate-dummy-data/src/com/specmate/dummydata/DummyProject.java (hunk 2 above)
Comment (junkerm) [reply bundles/specmate-dummy-data/src/com/specmate/dummydata/DummyProject.java]: done

File: bundles/specmate-jira-connector/src/com/specmate/connectors/jira/internal/services/JiraExportService.java (hunk 3 above)
Comment (junkerm) [reply bundles/specmate-jira-connector/src/com/specmate/connectors/jira/internal/services/JiraExportService.java]: done

File: bundles/specmate-jira-connector/src/com/specmate/connectors/jira/internal/services/JiraUtil.java (hunk 4 above)
Comment (junkerm) [reply bundles/specmate-jira-connector/src/com/specmate/connectors/jira/internal/services/JiraUtil.java]: done

File: bundles/specmate-dummy-data/src/com/specmate/dummydata/DummyProject.java (hunk 1 above)
Comment (junkerm) [reply bundles/specmate-dummy-data/src/com/specmate/dummydata/DummyProject.java]: done

**Review by sebeder — APPROVED** [review]

**Pull request merged**